win32ss/user/ntuser/ime.c

```diff
@@ -246,7 +250,11 @@ VOID UserFreeInputContext(PVOID Object)
 BOOLEAN UserDestroyInputContext(PVOID Object)
 {
     PIMC pIMC = Object;
-    UserDeleteObject(pIMC->head.h, TYPE_INPUTCONTEXT);
+    if (pIMC)
+    {
+        UserMarkObjectDestroy(pIMC);
+        UserDeleteObject(pIMC->head.h, TYPE_INPUTCONTEXT);
+    }
```

Comment:
Out of curiosity, when you test this code, do you know in which order, and when, these `UserFreeInputContext` and `UserDestroyInputContext` functions are called? This could help knowing the subtle difference between the object-free and object-destroy functionality in win32k.